lib/shared/addon/mixins/catalog-upgrade.js

```diff
@@ -0,0 +1,69 @@
+import Mixin from '@ember/object/mixin';
```

Comment:
There is a mixin `upgrade-component` that could possibly be used here. It would take some modification for your action but it does already have status and parsed versions and version links. If it doesn't make sense to use that rather then create a new mixin, is there a reason we're re-exporting it out of the engine in the corresponding app file? It doesn't look like its used outside of the monitoring engine so if it is not I'd like to remove the re-export so we're not bulking up our dist with unnecessary plumbing required for re-export.